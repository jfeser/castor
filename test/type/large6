select([sum((l_extendedprice * (1 - l_discount))) as revenue],
     atuple([select([l_extendedprice, l_discount],
               aorderedidx(dedup(
                             select([l_quantity],
                               join(((p_partkey = l_partkey) && true),
                                 filter(((l_shipinstruct = "DELIVER IN PERSON")
                                        &&
                                        ((l_shipmode = "AIR") ||
                                        (l_shipmode = "AIR REG"))),
                                   lineitem),
                                 filter((p_size >= 1), part)))),
                 alist(select([l_extendedprice, l_discount, p_brand],
                         join(((((p_container = "SM CASE") ||
                                ((p_container = "SM BOX") ||
                                ((p_container = "SM PACK") ||
                                (p_container = "SM PKG")))) &&
                               ((p_size <= 5) && (l_quantity = 0.l_quantity)))
                              && ((p_partkey = l_partkey) && true)),
                           filter(((l_shipinstruct = "DELIVER IN PERSON") &&
                                  ((l_shipmode = "AIR") ||
                                  (l_shipmode = "AIR REG"))),
                             lineitem),
                           filter((p_size >= 1), part))),
                   filter((p_brand = ""),
                     atuple([ascalar(0.l_extendedprice),
                             ascalar(0.l_discount), ascalar(0.p_brand)],
                       cross))),
                 >= 0, <= (0 + 10))),
             alist(select([l_quantity, l_extendedprice, l_discount, p_brand],
                     join(((((p_container = "MED BAG") ||
                            ((p_container = "MED BOX") ||
                            ((p_container = "MED PKG") ||
                            (p_container = "MED PACK")))) && (p_size <= 10)) &&
                          ((p_partkey = l_partkey) && true)),
                       filter(((l_shipinstruct = "DELIVER IN PERSON") &&
                              ((l_shipmode = "AIR") || (l_shipmode = "AIR REG"))),
                         lineitem),
                       filter((p_size >= 1), part))),
               filter(((p_brand = "") &&
                      ((l_quantity >= 0) && (l_quantity <= (0 + 10)))),
                 atuple([ascalar(0.l_quantity), ascalar(0.l_extendedprice),
                         ascalar(0.l_discount), ascalar(0.p_brand)],
                   cross))),
             alist(select([l_quantity, l_extendedprice, l_discount, p_brand],
                     join(((((p_container = "LG CASE") ||
                            ((p_container = "LG BOX") ||
                            ((p_container = "LG PACK") ||
                            (p_container = "LG PKG")))) && (p_size <= 15)) &&
                          ((p_partkey = l_partkey) && true)),
                       filter(((l_shipinstruct = "DELIVER IN PERSON") &&
                              ((l_shipmode = "AIR") || (l_shipmode = "AIR REG"))),
                         lineitem),
                       filter((p_size >= 1), part))),
               filter(((p_brand = "") &&
                      ((l_quantity >= 0) && (l_quantity <= (0 + 10)))),
                 atuple([ascalar(0.l_quantity), ascalar(0.l_extendedprice),
                         ascalar(0.l_discount), ascalar(0.p_brand)],
                   cross)))],
       concat))
