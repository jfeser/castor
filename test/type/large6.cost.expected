(Ok -4611686018427387904)
